tests/testthat/test-score_forecasts.R

```diff
@@ -391,6 +391,1366 @@ test_that("abs error is correct, point and median forecasts same", {
+    location = locations,
+    value = y,
+    stringsAsFactors = FALSE
+  )
+  n_forecasts <- length(forecast_quantiles)
+  test_forecasts <- data.frame(
+    model = rep("m1", n_forecasts),
+    forecast_date = rep(lubridate::ymd("20200101"), n_forecasts),
+    location = forecast_locations,
+    horizon = forecast_horizons,
+    temporal_resolution = rep("wk", n_forecasts),
+    target_variable = forecast_target_variables,
+    target_end_date = forecast_target_end_dates,
+    type = rep("quantile", n_forecasts),
+    quantile = forecast_quantile_probs,
+    value = forecast_quantiles,
+    stringsAsFactors = FALSE
+  )
+  actual <- score_forecasts(
+    forecasts = test_forecasts, truth = test_truth,
+    use_median_as_point = TRUE
+  )
+  expected <- rep(0, length(y))
+  expect_equal(actual$sharpness, expected)
+})
+
+test_that("sharpness is correct, median only, use_median_as_point FALSE", {
+  y <- c(1, -15, 22)
+  forecast_quantiles_matrix <- rbind(
+    c(-1, 0, 1, 2, 3),
+    c(-2, 1, 2, 2, 4),
+    c(-2, 0, 3, 3, 4)
+  )
+  forecast_quantile_probs <- c(0.1, 0.25, 0.5, 0.75, 0.9)
+  forecast_quantiles_matrix <- forecast_quantiles_matrix[, 3, drop = FALSE]
+  forecast_quantile_probs <- forecast_quantile_probs[3]
+
+  target_end_dates <- lubridate::ymd(20200101) + c(7, 14, 7)
+  horizons <- c("1", "2", "1")
+  locations <- c("01", "01", "02")
+  target_variables <- rep("inc death", length(y))
+
+  forecast_target_end_dates <-
+    rep(target_end_dates, times = ncol(forecast_quantiles_matrix))
+  forecast_horizons <- rep(horizons, times = ncol(forecast_quantiles_matrix))
+  forecast_locations <- rep(locations, times = ncol(forecast_quantiles_matrix))
+  forecast_target_variables <-
+    rep(target_variables, times = ncol(forecast_quantiles_matrix))
+  forecast_quantile_probs <- rep(forecast_quantile_probs, each = length(y))
+  forecast_quantiles <- forecast_quantiles_matrix
+  dim(forecast_quantiles) <- prod(dim(forecast_quantiles))
+
+  test_truth <- data.frame(
+    model = rep("truth_source", length(y)),
+    target_variable = target_variables,
+    target_end_date = target_end_dates,
+    location = locations,
+    value = y,
+    stringsAsFactors = FALSE
+  )
+
+  n_forecasts <- length(forecast_quantiles)
+  test_forecasts <- data.frame(
+    model = rep("m1", n_forecasts),
+    forecast_date = rep(lubridate::ymd("20200101"), n_forecasts),
+    location = forecast_locations,
+    horizon = forecast_horizons,
+    temporal_resolution = rep("wk", n_forecasts),
+    target_variable = forecast_target_variables,
+    target_end_date = forecast_target_end_dates,
+    type = rep("quantile", n_forecasts),
+    quantile = forecast_quantile_probs,
+    value = forecast_quantiles,
+    stringsAsFactors = FALSE
+  )
+
+  expect_error(actual <- score_forecasts(
+    forecasts = test_forecasts, truth = test_truth,
+    use_median_as_point = FALSE
+  ))
+})
+
+test_that("sharpness is correct, 1 interval only, use_median_as_point TRUE", {
```

Comment:
This test seems to have two intervals in the data, although maybe only one is tested? seems like there should be alignment between the data passed and the test.

Reply:
The finalized data is only 1 interval alpha=0.2 so quantiles 0.1 and 0.9. I tried to remain consistent with the set up of wis.